docs/spec.md

```diff
@@ -140,240 +43,355 @@ stablehlo.func @main(
 }
 ```
 
-## Execution
+### Functions
 
-### Sequential execution
+```ebnf
+Func        ::= 'stablehlo' '.' 'func' FuncId FuncInputs FuncOutputs '{' FuncBody '}'
+FuncInputs  ::= '(' [FuncInput {',' FuncInput}] `)`
+FuncInput   ::= '%' ValueId ':' ValueType
+FuncOutputs ::= ['->' FuncOutput, {',' FuncOutput}]
+FuncOutput  ::= ValueType
+FuncBody    ::= {Op}
```

Comment:
A notation section would be great, saw in separate thread that it is underway which is great.

In the meantime, does `{ Thing }` mean zero-or-more, or one-or-more?

We allow empty programs `Program ::= Func*`, we do not allow empty function bodies `FuncBody ::= Op+`, or maybe `FuncBody ::= {Op} ReturnOp` may be more accurate if we want to keep the current notation.

_Unless_ we are capturing the language of unverified programs, then this is accurate.

Reply:
Just seeing below that we have `digit {digit}` which makes me think it means zero-or-more. In that case I'd suggest the return op change